packages/web-forms/src/components/common/map/useMapBlock.ts

```diff
@@ -0,0 +1,293 @@
+import {
+	getSavedStyles,
+	getSelectedStyles,
+	getUnselectedStyles,
+} from '@/components/common/map/map-styles.ts';
+import type { FeatureCollection } from 'geojson';
+import { Map, MapBrowserEvent, View } from 'ol';
+import { Zoom } from 'ol/control';
+import Feature from 'ol/Feature';
+import GeoJSON from 'ol/format/GeoJSON';
+import { LineString, Point, Polygon } from 'ol/geom';
+import TileLayer from 'ol/layer/Tile';
+import WebGLVectorLayer from 'ol/layer/WebGLVector';
+import type { Pixel } from 'ol/pixel';
+import { fromLonLat } from 'ol/proj';
+import { OSM } from 'ol/source';
+import VectorSource from 'ol/source/Vector';
+import { computed, ref, shallowRef, watch } from 'vue';
+
+type GeometryType = LineString | Point | Polygon;
+
+const STATES = {
+	LOADING: 'loading',
+	READY: 'ready',
+	ERROR: 'error',
+} as const;
+
+const DEFAULT_GEOJSON_PROJECTION = 'EPSG:4326';
+const DEFAULT_VIEW_CENTER = [0, 0];
+const MAX_ZOOM = 16;
+const MIN_ZOOM = 2;
+const GEOLOCATION_TIMEOUT_MS = 10 * 1000;
+const ANIMATION_TIME = 1000;
+const SMALL_DEVICE_WIDTH = 576;
+const FEATURE_ID_PROPERTY = 'odk_feature_id';
+const SAVED_ID_PROPERTY = 'savedId';
+const SELECTED_ID_PROPERTY = 'selectedId';
+
+export function useMapBlock() {
+	const currentState = shallowRef<(typeof STATES)[keyof typeof STATES]>(STATES.LOADING);
+	const errorMessage = shallowRef<{ title: string; message: string } | undefined>();
+	let mapInstance: Map | undefined;
+	const savedFeature = ref<Feature<GeometryType> | undefined>();
+	const selectedFeature = ref<Feature<GeometryType> | undefined>();
```

Comment:
Could these be shallow refs? Do you ever mutate a `Feature` object after setting it on the ref?